homeassistant/components/snips/__init__.py

```diff
@@ -92,21 +91,16 @@
 async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
     """Activate Snips component."""
 
+    await mqtt.async_wait_until_loaded(hass)
```

Comment:
Wow.  Much better!

Reply:
But not good enough. I'll try tomorrow to finish the PR. Calling `hass.async_block_till_done()` would wait for ever.

Reply:
Ah, that's true.  I think you can use instead the common pattern of raising not ready yet and letting Home Assistant retry the load in the background for you.

Reply:
Now we have `hass.config_entries.async_wait_for_states()` we can await the entry setup.